front/lib/resources/base_resource.ts

```diff
@@ -28,7 +30,7 @@ export abstract class BaseResource<M extends Model> {
     this.id = blob.id;
   }
 
-  static async fetchById<T extends BaseResource<M>, M extends Model>(
+  static async fetchByInternalId<T extends BaseResource<M>, M extends Model>(
```

Comment:
If we rename it I would call it `fetchByModelId` to be even more explicit

Reply:
Fair point.